zio-cache/shared/src/main/scala/zio/cache/ManagedCache.scala

```diff
@@ -0,0 +1,388 @@
+package zio.cache
+
+import zio.ZManaged.Finalizer
+import zio._
+import zio.internal.MutableConcurrentQueue
+
+import java.time.{Clock, Duration, Instant}
+import java.util
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicInteger, LongAdder}
+import scala.jdk.CollectionConverters._
+
+sealed abstract class ManagedCache[-Key, +Error, +Value] {
+
+  /**
+   * Returns statistics for this cache.
+   * @return
+   */
+  def cacheStats: UIO[CacheStats]
+
+  /**
+   * Return whether a Managed associated with the specified key exists in the cache.
+   * @param key
+   * @return
+   */
+  def contains(key: Key): UIO[Boolean]
+
+  /**
+   * Return statistics for the specified entry.
+   */
+  def entryStats(key: Key): UIO[Option[EntryStats]]
+
+  /**
+   * Give a proxy managed on the resource already created for this key if any.
+   * Otherwise computes the managed with the lookup function, puts it in the cache,
+   * and returns it.
+   * The given managed is lazy, before it's actually used the resource is not created and
+   * cached.
+   * The register creation time for the given resource is saved one the first managed used
+   * @param key
+   * @return
+   */
+  def get(key: Key): UIO[Managed[Error, Value]]
```

Comment:
I think this should return a `Managed[Error, Value]` instead of a `UIO[Error, Value]`.

Reply:
Yes, I thought about that before and tried to move to `Managed[Error, Value]` but broke a lot of tests.
I tried again and get something working now! Made that change in a new commit